### DISCUSSION TITLE
Artificial Intelligence (AI) Integrations

### DISCUSSION THREAD

**Discussion by charlesweng** [stride3d/stride#2316] [Q&A]
Does anybody have any ideas on how to integrate AI or other features into the game engine? So far, I can only think of converting the game to Unity and then integrating ML-Agents (Unity-based AI) to the game?

**Comment by VaclavElias** [comment]
One idea, you can use https://devblogs.microsoft.com/dotnet/openai-dotnet-library/ for any .NET application 🙂.

**Comment by fydar** [comment]
You can also use ML.NET, which is the Microsoft official AI offering for .NET:

https://dotnet.microsoft.com/en-us/apps/machinelearning-ai/ml-dotnet

**Comment by Nicogo1705** [comment]
you can also take a look at : https://github.com/SciSharp/LLamaSharp

Small models can run locally with your game on a lot of computer nowdays.

**Comment by delustra** [comment]
check this out for reinforcement learning https://github.com/asieradzk/RL_Matrix

**Comment by VaclavElias** [comment]
Additional learning links:

https://learn.microsoft.com/en-gb/dotnet/ai/
https://devblogs.microsoft.com/dotnet/build-gen-ai-with-dotnet-8/
